app/services/data_migrations/remove_surplus_reference_selections.rb

```diff
@@ -0,0 +1,25 @@
+module DataMigrations
+  class RemoveSurplusReferenceSelections
+    TIMESTAMP = 20210616160559
+    MANUAL_RUN = false
+
+    def change
+      selected_reference_counts =
+        ApplicationForm.joins(:application_references)
+        .where(references: { selected: true })
+        .group('id')
+        .select('"application_forms".id, COUNT("references".id) AS ref_count')
+
+      applications_with_surplus_selections =
+        ApplicationForm.where(
+          id: selected_reference_counts.select { |c| c.ref_count > 2 }.map(&:id),
+        )
+
+      applications_with_surplus_selections.each do |application|
```

Comment:
```suggestion
      applications_with_surplus_selections.find_each do |application|
```

Though it won't matter if there are only 90 records in the scope.